manim/plugins/import_plugins.py

```diff
@@ -0,0 +1,41 @@
+import pkg_resources
+from importlib import import_module
+from .. import config, logger
+
+__all__ = []
+
+# don't know what is better to do here!
+module_type = type(pkg_resources)
+function_type = type(import_module)
+
+plugins_requested: list = config["plugins"]
+if "" in plugins_requested:
+    plugins_requested.remove("")
+for plugin in pkg_resources.iter_entry_points("manim.plugins"):
+    if plugin.name in plugins_requested:
+        loaded_plugin = plugin.load()
+        if isinstance(loaded_plugin, module_type):
+            # it is a module so it can't be called
+            # see if __all__ is defined
+            # if it is defined use that to load all the modules necessary
+            # essentially this would be similar to `from plugin import *``
```

Comment:
Do the plugins' namespaces have to be eradicated? Because I'd rather they be preserved

Reply:
I would like that to happen :)
Because users can also import plugins manually if there want its namespace.

Reply:
I think the result of the discussion on Discord was that we should use a config file to determine the default plugins that are imported/used (as well as manually using `--plugins` to specify which plugins to use/import in the entry point)

Reply:
Yeah, I'm just not a fan of the possible namespace thrashing

Reply:
I don't mind the potential namespace thrashing (some plugins like the manim-onlinetex @Aathish04 rely on this); however, I do mind the syntax highlighting issue this presents and I think others would too -- which is why I would opt for CLI to change the `plugins/__init__.py` file which wouldn't hinder syntax highlighting

Reply:
> for CLI to change the plugins/__init__.py file which wouldn't hinder syntax highlighting

That is bad practice. Some package managers checksum the files it installs and doing so will break it. And other times things installed from the package manager isn't usually writable and it would need to run in sudo mode which may create unnecessary problems.